Branch selector styling adjustments

Pat-Stuart opened the pull request:
[Relevant PBI](https://github.com/tinacms/tinacms/issues/5935)

This PR makes adjustments to the styling of the current branch selector to align with changes to the collections table. It ensures:

- Alignment of all content, regardless of protected branch status, 
- Semantic correctness of HTML elements (Move from divs to a table).
- No repetition of headers

From:

<img width="844" height="448" alt="image" src="https://github.com/user-attachments/assets/5f8a897e-2759-4605-b951-6c4e6f87640e" />

**Figure: Current branch selector**

<img width="462" height="726" alt="image" src="https://github.com/user-attachments/assets/9214ce17-28f3-4c47-8ff8-66fd0f5ee9eb" />

**Figure: Current Mobile view**

To: 

<img width="973" height="487" alt="image" src="https://github.com/user-attachments/assets/d25ca794-7a39-4178-8425-b4782a54a033" />

**Figure: New branch selector styling**

<img width="424" height="730" alt="image" src="https://github.com/user-attachments/assets/55cb539b-2723-4e51-9338-2b3ee6087408" />

**Figure: Mobile styling with scrollable table**


<!--

Thank you for contributing to TinaCMS!

Several things must happen for your PR to be merged:

1. It must successfully build, test, and lint your branch
1. It must pass the dangerjs script
2. It must be up-to-date with master
3. It must be approved by at least 1 tinacms core member
4. It must have conventional-commits that clearly explain what has been changed


Small is beautiful! 

PRs should be small. They should be made up of many small commits, with clear 
commit messages explaining _what_ has been changed and _why_. The tests should
be short and specific, with single assertions. 

-->

joshbermanssw commented on packages/tinacms/src/toolkit/plugin-branch-switcher/branch-switcher.tsx:
```diff
@@ -425,18 +428,30 @@ const BranchSelector = ({
+          <table className='w-full'>
+            <thead className='bg-white border-b-2 border-gray-100'>
+              <tr>
+                <th className={`${tableHeadingStyle}`}>Branch Name</th>
```
```suggestion
                <th className={tableHeadingStyle}>Branch Name</th>
```

Pat-Stuart replied:
✅ Done in https://github.com/tinacms/tinacms/pull/6130/commits/ba4333e55b84684792bd94afcb6fabd6bf07c4bd

Pat-Stuart commented:
Update - UI Test Passed by @micsblank 

Changes have been made to align with UI best practices:

- Aligning labels and inputs in branch selector
- Background color for header
- Badge added to currently selected branch
- Header text changed from uppercase
- CollectionsListPage was changed to align with this new style

<img width="1920" height="916" alt="image" src="https://github.com/user-attachments/assets/afa21598-5eec-4f2f-b3f5-b6c1bf3daa3f" />

**Figure: Collections page with style adjustments**

<img width="835" height="559" alt="image" src="https://github.com/user-attachments/assets/b0d44a8f-25c7-4687-9995-a11a6325eb65" />

**Figure: Updated branch selector**

<img width="430" height="711" alt="image" src="https://github.com/user-attachments/assets/f59963e9-80cb-451d-9660-d37b5824c36f" />

**Figure: Updated branch selector in Mobile UI**
